e2e/modules/environment.js

```diff
@@ -243,30 +243,46 @@ module.exports = {
 
     async getServerMap(app) {
         const map = {};
-        await Promise.all(app.windows().map(async (win) => {
-            return win.evaluate(async () => {
-                if (!window.testHelper) {
-                    return null;
-                }
-                const info = await window.testHelper.getViewInfoForTest();
-                return {viewName: `${info.serverName}___${info.viewType}`, webContentsId: info.webContentsId};
-            }).then((result) => {
-                if (result) {
-                    map[result.viewName] = {win, webContentsId: result.webContentsId};
-                }
-            });
-        }));
+        await Promise.all(app.windows().
+            filter((win) => !win.url().includes('mattermost-desktop://')).
+            map(async (win) => {
+                return win.evaluate(async () => {
+                    if (!window.testHelper) {
+                        return null;
+                    }
+                    const info = await window.testHelper.getViewInfoForTest();
+                    return {viewName: `${info.serverName}___${info.viewType}`, webContentsId: info.webContentsId};
+                }).then((result) => {
+                    if (result) {
+                        map[result.viewName] = {win, webContentsId: result.webContentsId};
+                    }
+                });
+            }));
         return map;
     },
 
     async loginToMattermost(window) {
-        // Do this twice because sometimes the app likes to load the login screen, then go to Loading... again
         await asyncSleep(1000);
         await window.waitForSelector('#input_loginId');
         await window.waitForSelector('#input_password-input');
         await window.waitForSelector('#saveSetting');
 
-        await window.type('#input_loginId', process.env.MM_TEST_USER_NAME);
+        let username;
+        switch (process.platform) {
+        case 'darwin':
+            username = 'success+sysadmin+macos@simulator.amazonses.com';
+            break;
+        case 'linux':
+            username = 'success+sysadmin+linux@simulator.amazonses.com';
+            break;
+        case 'win32':
+            username = 'success+sysadmin+windows@simulator.amazonses.com';
+            break;
+        default:
+            throw new Error('Unsupported platform');
+        }
```

Comment:
Why do we need to hardcode these usernames and for each platform?

Reply:
Hi @saturninoabril 
This is a temp fix untill we have a support to spin up a server on run time and destroying it.

Since we use same user account and 1 instance, we end up running  same tests in parallel on 3 OS. There are quite few tests that fail as they interact with default channel or slash command or assert text posted during other OS test is running. To avoid that I have created these on the `/cloud` instance we use to test.